### PR TITLE
Frontend: MainView & LogInView

### PR DESCRIPTION
**Pull request by Kr0nox**
This PR implements the MainView and the LogInView.
In addition it adds the favicon to the page tab.

Will be converted to from a draft to a PR when the restInterfaces are implemented.

### CODE REVIEW

File: frontend/index.html [hunk 1]
```diff
@@ -2,7 +2,7 @@
 <html lang="en">
     <head>
         <meta charset="UTF-8" />
-        <link rel="icon" type="image/svg+xml" href="/vite.svg" />
+        <link rel="icon" type="image/png" href="/favico.png" />
```
Comment (c0derMo) [inline frontend/index.html]: typo in favicon filename

File: frontend/src/components/ButtonComponent.vue [hunk 2]
```diff
@@ -1,13 +1,39 @@
...
-  </InteractableComponent>
+  <div>
+    <div class="relative">
+      <div class="absolute w-full h-full text-center" v-if="loading">
+        <div
+          class="absolute h-full w-full bg-gray-300 opacity-70 rounded-md"
+        ></div>
+        <FontAwesomeIcon icon="spinner" size="1x" class="animate-spin mt-2" />
+      </div>
+      <InteractableComponent @click="(e) => $emit('click', e)">
```
Comment (c0derMo) [inline frontend/src/components/ButtonComponent.vue:12]: We could add some functionality that prevents click events while the spinner is showing - this way we might avoid accidential double-presses or similar - Opinions?

**Review by c0derMo — CHANGES_REQUESTED** [review]

File: frontend/src/components/ButtonComponent.vue [hunk 3]
```diff
@@ -1,13 +1,39 @@
...
-  <InteractableComponent @click="(e) => $emit('click', e)">
-    <!-- Slot for the text of the button -->
-    <slot></slot>
-  </InteractableComponent>
+  <div>
+    <div class="relative">
+      <div class="absolute w-full h-full text-center" v-if="loading">
+        <div
+          class="absolute h-full w-full bg-gray-300 opacity-70 rounded-md"
+        ></div>
```
Comment (Kr0nox) [inline frontend/src/components/ButtonComponent.vue]: What is this empty div for?

File: frontend/src/components/CodeInput.vue [hunk 4]
```diff
@@ -0,0 +1,54 @@
+<template>
+  <TextInputComponent
+    uppercase
```
Comment (Kr0nox) [inline frontend/src/components/CodeInput.vue]: I would write `:uppercase=true` here to make clear it is a prop

File: frontend/src/components/CodeInput.vue [hunk 5]
```diff
@@ -0,0 +1,54 @@
+<template>
+  <TextInputComponent
```
Comment (Kr0nox) [inline frontend/src/components/CodeInput.vue]: during loading typing should be forbidden

File: frontend/src/components/ContainerComponent.vue [hunk 6]
```diff
@@ -1,7 +1,7 @@
 <!-- Main component for seperating semantic contents -->
 <template>
   <div
-    class="p-2 border-1 rounded-md bg-container border-container-border box-border shadow overflow-hidden flex flex-col max-h-0 min-h-full"
```
Comment (Kr0nox) [inline frontend/src/components/ContainerComponent.vue:4]: was the component tested after this was removed?

File: frontend/src/views/LogInView.vue [hunk 7]
```diff
@@ -0,0 +1,100 @@
...
+        <h1 class="underline text-xl">Anmelden:</h1>
+      </template>
+      <template #default>
+        <table class="w-full">
+          <tr>
+            <td class="p-1">Nutzername/Email:</td>
+            <td class="w-full p-1">
+              <TextInputComponent
+                placeHolder="Nutzername/Email"
+                @input-changed="(val) => (mailField = val)"
```
Comment (Kr0nox) [inline frontend/src/views/LogInView.vue:17]: this should be done over v-model

File: frontend/src/views/LogInView.vue [hunk 8]
```diff
@@ -0,0 +1,100 @@
...
+              />
+            </td>
+          </tr>
+          <tr>
+            <td class="p-1">Passwort:</td>
+            <td class="w-full p-1">
+              <TextInputComponent
+                :hidden="true"
+                placeHolder="Passwort"
+                @input-changed="(val) => (passwordField = val)"
```
Comment (Kr0nox) [inline frontend/src/views/LogInView.vue:28]: this should be done over v-model

File: frontend/src/views/LogInView.vue [hunk 9]
```diff
@@ -0,0 +1,100 @@
...
+              v-model="keepLoggedIn"
+            />
+            <label class="cursor-pointer" @click="keepLoggedIn = !keepLoggedIn"
+              >Angemeldet bleiben</label
+            >
+          </div>
+          <div class="flex-grow">
+            <!-- Empty div as space filler -->
+          </div>
+          <div class="w-fit text-red-500 italic" v-if="displayError">
```
Comment (Kr0nox) [inline frontend/src/views/LogInView.vue]: use the above div for this and remove the space filler below

File: frontend/src/views/LogInView.vue [hunk 10]
```diff
@@ -0,0 +1,100 @@
...
+const keepLoggedIn = ref(false);
+const displaySpinner = ref(false);
+const displayError = ref(false);
+
+/**
+ * Checks whether the login data is correct
+ */
+async function checkLogIn(): Promise<void> {
+  //avoid multiple clicks on button
+  if (displaySpinner.value) {
```
Comment (Kr0nox) [inline frontend/src/views/LogInView.vue]: during this all inputs should be disabled

File: frontend/src/views/MainView.vue [hunk 11]
```diff
@@ -1,9 +1,70 @@
...
-    <!-- Empty div so eslint is happy -->
+  <div class="flex flex-col">
+    <img src="../assets/Logo.png" class="w-1/3 mx-auto" />
+    <ContainerComponent class="mx-auto space-y-2 px-10 py-3">
+      <template #Header>
+        <h1 class="text-3xl text-center">Code eingeben:</h1>
+      </template>
+      <template #default>
+        <div class="relative">
+          <div class="absolute w-full h-full text-center" v-if="displaySpinner">
```
Comment (Kr0nox) [inline frontend/src/views/MainView.vue]: when the spinner is displayed the codeinput should not be typeable

**Review by Kr0nox — COMMENTED** [review]
Some things I found

File: frontend/src/components/ButtonComponent.vue (hunk 3 above)
Comment (a-bumblebee) [reply frontend/src/components/ButtonComponent.vue]: This is the 70-%-transparent background of the overlay behind the spinner. It is used to lift the spinner from the text inside the button.

Comment (Kr0nox) [reply frontend/src/components/ButtonComponent.vue]: why isnt the spinner inside this div then? or the background property just on the spinner?

Comment (a-bumblebee) [reply frontend/src/components/ButtonComponent.vue]: Then the spinner would also be 70%-transparent

Comment (Kr0nox) [reply frontend/src/components/ButtonComponent.vue]: also if the opacity was changed to the class `bg-opacity-70`?

Comment (a-bumblebee) [reply frontend/src/components/ButtonComponent.vue]: Maybe, I'll try

Comment (a-bumblebee) [reply frontend/src/components/ButtonComponent.vue]: Yes, that works, I'll change that

File: frontend/src/components/ButtonComponent.vue (hunk 2 above)
Comment (Kr0nox) [reply frontend/src/components/ButtonComponent.vue:12]: either this or just implement not calling the coresponding method in the view twive

Comment (c0derMo) [reply frontend/src/components/ButtonComponent.vue:12]: I feel like disabling while spinning is the much easier approach - not calling twice requires checking timings between presses, etc.

File: frontend/src/views/MainView.vue [hunk 12]
```diff
@@ -1,9 +1,70 @@
...
+            <div
+              class="absolute h-full w-full bg-gray-300 opacity-70 rounded-md"
+            ></div>
+            <FontAwesomeIcon
```
Comment (c0derMo) [inline frontend/src/views/MainView.vue]: I'd probably also move the spinner inside CodeInput

File: frontend/src/components/ContainerComponent.vue (hunk 6 above)
Comment (a-bumblebee) [reply frontend/src/components/ContainerComponent.vue:4]: Wasn't this your change? None the less no bugs have occured so far.

Comment (Kr0nox) [reply frontend/src/components/ContainerComponent.vue:4]: I cant remember. If it neds to be cahnged back somewhere we will just do so in that PR

File: frontend/src/views/LogInView.vue (hunk 8 above)
Comment (Kr0nox) [reply frontend/src/views/LogInView.vue:28]: shouldn't. my bad

File: frontend/src/views/LogInView.vue (hunk 7 above)
Comment (Kr0nox) [reply frontend/src/views/LogInView.vue:17]: shouldn't. my bad

**Review by c0derMo — APPROVED** [review]

File: frontend/src/components/CodeInput.vue [hunk 13]
```diff
@@ -0,0 +1,73 @@
...
+defineProps({
+  /**
+   * Display red border line
+   */
+  showRedBorder: {
+    type: Boolean,
+    required: true,
+  },
+
+  disableInputShowSpinner: {
```
Comment (Kr0nox) [inline frontend/src/components/CodeInput.vue:41]: JSDoc missing

File: frontend/src/components/CodeInput.vue [hunk 14]
```diff
@@ -0,0 +1,73 @@
+<template>
```
Comment (Kr0nox) [inline frontend/src/components/CodeInput.vue:2]: Doc comment missing

File: frontend/src/views/LogInView.vue [hunk 15]
```diff
@@ -0,0 +1,104 @@
+<template>
```
Comment (Kr0nox) [inline frontend/src/views/LogInView.vue:2]: JSDoc comment

**Review by a-bumblebee — APPROVED** [review]

**Review by c0derMo — APPROVED** [review]

**Pull request merged**